frontend/src/components/Modal.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { useEffect } from 'react';
+import ReactDOM from 'react-dom';
+import { useModalStack } from '@/context/ModalContext';
+import { useModal } from '@/hooks/useModal';
+
+type ModalProps = {
+  children: React.ReactNode;
+  id: string;
+  isOpen: boolean;
+  onClose: () => void;
+};
+
+export const Modal: React.FC<ModalProps> = ({
+  id,
+  isOpen,
+  onClose,
+  children,
+}) => {
+  // eslint-disable-next-line react/hook-use-state
```

Comment:
Given that i was on a rush, and was not sure how much i should respect the original configs lint rules i had to disable some rules here and there, i know it should be clean.... but i was trying to get the code out. :)